src/widgets/Console.ts

```diff
@@ -0,0 +1,91 @@
+import { v, w } from '@dojo/widget-core/d';
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import { WidgetProperties } from '@dojo/widget-core/interfaces';
+import { ThemedMixin, ThemedProperties, theme } from '@dojo/widget-core/mixins/Themed';
+import { stringify } from '../util';
+import ActionBar, { ActionBarButton } from './ActionBar';
+import * as css from '../styles/console.m.css';
+
+export enum ConsoleMessageType {
+	Log = 'log',
+	Error = 'error',
+	Info = 'info',
+	Warning = 'warn'
+}
+
+export interface ConsoleMessage {
+	type: ConsoleMessageType;
+	message: any | any[];
```

Comment:
Is there a reason this can't be more narrow than `any`?

Reply:
And technically, `any | any[]` gets reduced to `any`... (which is why it scares me)

Reply:
I can change it to be `any` and then always check and convert to an array, which is what I think I am doing. I'm not sure if it should be something other than `any`, though. Per TypeScript's [types](https://github.com/Microsoft/TypeScript/blob/8f23bf813c249a1502ed12e58a775e5ed6d25e40/lib/lib.d.ts#L5873), `console.log`  and the other methods accept `any`.

```ts
log(message?: any, ...optionalParams: any[]): void;
```

Reply:
👍 I didn't clock it was mirroring the log method...  I guess `any | any[]` does help with the clarity, though if you are just dumping the args in there, wouldn't it always be `any[]`?

Since it is an exported interface, a bit of JSDoc on each of the properties would help clarity.